File: c#/program6.cs
using System;

class Factorial
{
    public int No;

    public Factorial(int no)
    {
        this.No = no;
    }

    public int DisplayFactorial()
    {
        int iFact = 1;

        for(int iCnt = 1; iCnt <= No; iCnt++)
        {
            iFact = iFact * iCnt;  
        }
        return iFact;
    }
}

class program6
{
    public static void Main()
    {
        int value = 0;
        int ret = 0;

        Console.WriteLine("Enter number : ");
        value = int.Parse(Console.ReadLine());

        Factorial obj = new Factorial(value);
        ret = obj.DisplayFactorial();

        Console.WriteLine("Factorial is : "+ret);
    }
}

File: cpp/program6.cpp
// program42.c

#include<iostream>
using namespace std;

typedef unsigned long int ULINT;

class Factorial
{
    public:
       int iNo;

    Factorial(int X)
    {
        iNo = X;
    }

    ULINT DisplayFactorial()
    {
        int iCnt = 0;
        ULINT iFact = 1;

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        for(iCnt = 1; iCnt <= iNo; iCnt++)
        {
            iFact = iFact * iCnt;
        }
        return iFact;
    }
};

int main()
{
    int iValue = 0;
    ULINT iRet = 0;

    cout<<"Enter number"<<"\n";
    cin>>iValue;

    Factorial fobj(iValue);
    iRet = fobj.DisplayFactorial();

    cout<<"Factorial is : "<<iRet<<"\n";

    return 0;

}